shared/metrics/src/main/java/io/pravega/shared/MetricsNames.java

```diff
@@ -247,4 +220,49 @@ public static String failMetricName(String metricName) {
             return metricName + "_fail";
         }
     }
+
+    /**
+     * Convenient method to join two Strings with dot.
+     *
+     * @param element1 the first String object.
+     * @param element2 the second String object.
+     * @return String object with element1 and element2 joined with dot.
+     */
+    public static String joinWithDot(String element1, String element2) {
```

Comment:
IMO, joinWithDot() doesn't look like representing a functional behavior of the entity `MetricNames` that this class represents. 

Is this for concatenating metric names? If so, it could perhaps be represented by the signature `concatenate(String metric1, String metric2)`. 

Or maybe, this is for concatenating two elements of different kinds and not metrics. Perhaps a better location for such a method is a utility class.

Reply:
Will said and agree with all the principals mentioned:
1. Agree the method should go to utility class. MetricsNames.java is actually a sort of utility class, just without the "Util" on the name. It is called "MetricsNames" instead of "MetricName" to follow the Java Utility class convention, such as Strings.
2. Also struggling with the method name "join" or "concatenate" because the name doesn't call out the "dot" in-between.
3. Trying to make a perfect balance, I tried to move this method into NameUtils.java or StringUtil.java, but due to the dependency relationship, metric related codes do not have access to these utility classes.

Hope this explains.

Reply:
Sounds good.